### DISCUSSION TITLE
Using upsert when querying many-to-many relations

### DISCUSSION THREAD

**Discussion by amirhhashemi** [prisma/prisma#9057] [Q&A] [answered]
I have a situation just like this example:
https://www.prisma.io/docs/support/help-articles/working-with-many-to-many-relations#implicit-relations

I have a post model that has a list of tags. I want to create a post and its tags at the same time but on the contrary of that example, I want to use the "upsert" method instead of the "create" method. because the tag name is the primary key and I don't want to create a tag that exists. this is what I'm trying to do:

```ts
     const tags = ["programming", "javascript"]
      const post = await prisma.post.update({
        where: { id },
        data: { title, description, body, cover, tags: {
          upsert: tags.map(tag => {
            return {
              where: {name: tag},
              update: {},
              create: {name: tag}
            }
          })
        }},
      });
```
apparently, this doesn't work. what is the right way to do that?

**Comment by ryands17** [comment, marked as answer]
You need to use `connectOrCreate` in this case and not `upsert` as shown here: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#connectorcreate

**Reply by amirhhashemi** [reply]
Thank you :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
